### PR TITLE
Added support to generate app schema automatically

### LINKED ISSUE #69: Generate app schema automatically
currently, Custom Text authoring resource doesn't have an api to retreive app schema
So what we're doing, is let the user figure out how to get all custom entities in the app and provide them in the schema.json file (see Docs folder)
So we need to generate the schema automatically
proposed solution:
use the 'id_**_labels.json' file in the blob container used by CustomText
this file contains all info about labeling
use the 'entityNames' section that contains the labels/entities

### PR DESCRIPTION
**Pull request by ZulaMostafa**
Closes #69.

This PR includes adding support to generate app schema automatically using custom text authoring api to pull entities from language resource

### CODE REVIEW

File: CustomTextAnalytics.CognitiveSearch/Solution/IndexingCliTool/CognitiveSearchIntegration.Core/Orchestrators/IndexingOrchestrator.cs [hunk 1]
```diff
@@ -19,33 +20,34 @@ public class IndexingOrchestrator
...
             var indexerName = indexName.ToLower() + Constants.IndexerSuffix;
             var skillSetName = indexName.ToLower() + Constants.SkillsetSuffix;
             var customTextSkillName = indexName.ToLower() + Constants.CustomSkillSuffix;
-            var schemaFileName = Path.GetFileName(schemaPath);
 
-            // read schema
-            _loggerService.LogOperation(OperationType.ReadingAppSchema, schemaFileName);
-            var schemaString = await _storageService.ReadAsStringFromAbsolutePathAsync(schemaPath);
-            var schema = JsonHandler.DeserializeObject<CustomTextSchema>(schemaString, schemaFileName);
+            // pull extractors
```
Comment (mshaban-msft) [inline CustomTextAnalytics.CognitiveSearch/Solution/IndexingCliTool/CognitiveSearchIntegration.Core/Orchestrators/IndexingOrchestrator.cs:48]: rename operation to reading app schema

File: CustomTextAnalytics.CognitiveSearch/Solution/IndexingCliTool/CognitiveSearchIntegration.Definitions/Consts/Constants.cs [hunk 2]
```diff
@@ -9,5 +9,7 @@ public class Constants
         public const string DataSourceSuffix = "-data";
         public const string CustomSkillSuffix = "-customtext-skill";
         public const string SkillsetSuffix = "-skillset";
+        public const string CustomTextAuthoringApiVersion = "2021-11-01-preview";
+        public const string CustomTextAuthoringApiVersionHeader = "api-version";
```
Comment (mshaban-msft) [inline CustomTextAnalytics.CognitiveSearch/Solution/IndexingCliTool/CognitiveSearchIntegration.Definitions/Consts/Constants.cs:13]: api-version is not a constant
leave it at service level

**Comment by mshaban-msft** [comment]
can you please update the readme file to show the new indexing command doesn't use --schema

**Review by mshaban-msft — APPROVED** [review]

**Comment by mshaban-msft** [comment]
please don't forget to test the changes
i.e. run indexer and try searching

**Comment by ZulaMostafa** [comment]
> please don't forget to test the changes i.e. run indexer and try searching

everything works well, I tested it one more time after doing the changes mentioned in the comments.

**Review by mshaban-msft — APPROVED** [review]

**Pull request merged**